modules/oe_whitelabel_extra_project/src/Plugin/ExtraField/Display/EuContributionExtraField.php

```diff
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_whitelabel_extra_project\Plugin\ExtraField\Display;
+
+use Drupal\Core\Entity\ContentEntityInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\extra_field\Plugin\ExtraFieldDisplayFormattedBase;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Display overall budget.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_whitelabel_extra_project_eu_contrib",
+ *   label = @Translation("EU contribution"),
+ *   bundles = {
+ *     "node.oe_project",
+ *   },
+ *   visible = true
+ * )
+ */
+class EuContributionExtraField extends ExtraFieldDisplayFormattedBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The entity view builder.
+   *
+   * @var \Drupal\Core\Entity\EntityViewBuilder
+   */
+  protected $viewBuilder;
+
+  /**
+   * OverallBudgetExtraField constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->viewBuilder = $entity_type_manager->getViewBuilder('node');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->t('EU contribution');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    if ($entity->get('oe_project_eu_contrib')->isEmpty() && $entity->get('oe_project_budget_eu')->isEmpty()) {
+      return [];
+    }
+    $build = [];
+
+    $display_options = [
```

Comment:
Instead of making these settings unchangeable unless we update here, what about just using the configuration of the display?
Is there a way to read the current field (e.g. "oe_project_eu_contrib") display settings and use those to render either the new one, or the old one?

Reply:
Something along these lines:
```
// @todo can we get the current view mode?
$display = EntityViewDisplay::collectRenderDisplay($entity, $view_mode);
$display_settings = $display->getComponent('oe_project_eu_contrib');
$field_name = $entity->get('oe_project_eu_contrib')->isEmpty() ? 'oe_project_budget_eu' : 'oe_project_eu_contrib');
$build[] = $this->viewBuilder->viewField($entity->get($field_name), $display_options);

return $build;
```

Reply:
we have $view_mode = $this->getViewMode(); from the extra field base class but we don't have any configuration to load when we do getComponent(), because the old fields are gone, and the extra fields are not configurable